File: src/components/ship/ship_systems/ship_systems.rs
use rand_derive2::RandGen;
use super::{shield::Shield, hull::Hull, phaser::Phaser, torpedo::Torpedo};

#[derive(PartialEq, Debug, RandGen)]
pub struct ShipSystems {
    pub shield: Shield,
    pub hull: Hull,
    pub phaser: Phaser,
    pub torpedo: Torpedo,
}

File: StarTrekExplorers/Components/Ship/Capabilities/ShipSystems.cs
using StarTrekExplorers.Components.Interfaces;

namespace StarTrekExplorersTests.Entities
{
    public class ShipSystems : IShipSystems
    {
        public IDamageDealer Phaser { get; } = new Phaser();
        public IDamageDealer Torpedo { get; } = new Torpedo();
        public IDamageTaker Shield { get; } = new Shield();
        public IDamageTaker Hull { get; } = new Hull();
    }
}